src/WebJobs.Script.WebHost/Security/KeyManagement/BaseSecretsRepository.cs

```diff
@@ -78,6 +86,11 @@ protected void OnChanged(object sender, FileSystemEventArgs e)
 
         public abstract Task<string[]> GetSecretSnapshots(ScriptSecretsType type, string functionName);
 
+        protected virtual string GetLogErrorMessage(string operation)
+        {
+            return string.Format(Resources.TraceDeafultSecretRepoError, operation);
```

Comment:
We don't want to use Resources. It's an expensive call during cold start and we are going to move all log strings to extensions:
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceLoggerExtension.cs

You can can create an extension for Security